machine/v1beta1/types_vsphereprovider.go

```diff
@@ -87,12 +87,72 @@ type NetworkSpec struct {
 	Devices []NetworkDeviceSpec `json:"devices"`
 }
 
+// AddressesFromPool is an IPAddressPool that will be used to create
+// IPAddressClaims for fulfillment by an external controller.
+type AddressesFromPool struct {
+	// group of the IP address pool type known to an external IPAM controller.
+	// +kubebuilder:example=externalipam.controller.io
```

Comment:
I've not used the `example` marker before, will have to work out what that does.

Normally we explain in prose, perhaps we can do both? (Especially since this is a raw extension)

```suggestion
	// group of the IP address pool type known to an external IPAM controller.
	// This should be a fully qualified domain name, for example, externalipam.controller.io.
	// +kubebuilder:example=externalipam.controller.io
```